File: app/src/main/java/se/gustavkarlsson/skylight/android/dagger/modules/definitive/ActivityModule.java
package se.gustavkarlsson.skylight.android.dagger.modules.definitive;

import android.app.Activity;

import dagger.Module;
import dagger.Provides;
import se.gustavkarlsson.skylight.android.dagger.scopes.ActivityScope;

@Module
public class ActivityModule {
	private final Activity activity;

	public ActivityModule(Activity activity) {
		this.activity = activity;
	}

	@Provides
	@ActivityScope
	Activity provideActivity() {
		return activity;
	}
}

File: app/src/main/kotlin/se/gustavkarlsson/skylight/android/dagger/modules/definitive/ActivityModule.kt
package se.gustavkarlsson.skylight.android.dagger.modules.definitive

import android.app.Activity

import dagger.Module
import dagger.Provides
import se.gustavkarlsson.skylight.android.dagger.scopes.ActivityScope

@Module
class ActivityModule(private val activity: Activity) {

    @Provides
    @ActivityScope
    fun provideActivity(): Activity {
        return activity
    }
}
